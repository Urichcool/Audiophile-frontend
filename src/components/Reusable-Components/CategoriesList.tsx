import { FC } from "react";
import Button3 from "./Buttons/Button3";
import ShadowIcon from "../../images/icons/ShadowIcon";

interface ICategoriesProps {
  isMobile?: boolean;
}

const CategoriesList: FC<ICategoriesProps> = ({ isMobile }) => {
  return (
    <>
      <ul className="categories-list">
        <li className="categories-list-item">
          <div className="categories-list-hadphones-image" />
          <div className="categories-list-icon">
            <ShadowIcon />
          </div>
          <h6 className="H6-manrope-bold categories-list-item-title">
            headphones
          </h6>
          <Button3 to={"headphones"} isMobile={isMobile} />
        </li>
        <li className="categories-list-item">
          <div className="categories-list-speakers-image" />
          <div className="categories-list-icon">
            <ShadowIcon />
          </div>
          <h6 className="H6-manrope-bold categories-list-item-title">
            speakers
          </h6>
          <Button3 to={"speakers"} isMobile={isMobile} />
        </li>
        <li className="categories-list-item">
          <div className="categories-list-earphones-image" />
          <div className="categories-list-icon">
            <ShadowIcon />
          </div>
          <h6 className="H6-manrope-bold categories-list-item-title">
            earphones
          </h6>
          <Button3 to={"earphones"} isMobile={isMobile} />
        </li>
      </ul>
    </>
  );
};

export default CategoriesList;
