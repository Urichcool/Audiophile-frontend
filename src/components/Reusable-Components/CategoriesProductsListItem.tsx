import { FC } from "react";
import Button1 from "./Buttons/Button1";
import { AnimationOnScroll } from "react-animation-on-scroll";

interface ICategoriesProductsListItemProps {
  previewImage: { mobile: string; tablet: string; desktop: string };
  isNew: boolean;
  name: string;
  description: string;
  isEven: boolean;
  id: string
}

const CategoriesProductsListItem: FC<ICategoriesProductsListItemProps> = ({
  previewImage,
  name,
  isNew,
  description,
  isEven,
  id
}) => {
  return (
    <li
      className={
        isEven
          ? "categories-products-list-item-even"
          : "categories-products-list-item"
      }
    >
      <div className="categories-products-list-item-picture-container">
        <AnimationOnScroll animateIn="animate__fadeIn" animateOnce offset={0}>
          <picture>
            <source
              srcSet={previewImage.desktop}
              media="(min-width: 1440px)"
              width={"270"}
              height={"300"}
            />
            <source
              srcSet={previewImage.desktop}
              media="(min-width: 768px)"
              width={"220"}
              height={"250"}
            />
            <img
              src={previewImage.desktop}
              alt={name}
              width={"160"}
              height={"180"}
            />
          </picture>
        </AnimationOnScroll>
      </div>
      <AnimationOnScroll
        animateIn={isEven ? "animate__fadeInLeft" : "animate__fadeInRight"}
        delay={500}
        animateOnce
        offset={0}
      >
        <div className="categories-products-list-item-content-container">
          {isNew && (
            <p className="Overline-manrope-regular categories-products-list-item-new-product-text">
              NEW PRODUCT
            </p>
          )}
          <h2 className="H2-manrope-bold categories-products-list-item-title">
            {name}
          </h2>
          <p className="Body-manrope-medium categories-products-list-item-description">
            {description}
          </p>
          <Button1 id={id} testId={"id-page-nav-button"} />
        </div>
      </AnimationOnScroll>
    </li>
  );
};

export default CategoriesProductsListItem;
