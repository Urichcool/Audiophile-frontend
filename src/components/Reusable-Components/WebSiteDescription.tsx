import { FC } from "react";


const WebSiteDescription:FC = () => {
  return (
    <div className="web-site-description-container">
      <div className="web-site-description-picture" />
      <div className="web-site-description-text-container">
        <h4 className="H4-manrope-bold web-site-description-title">
          Bringing you the{" "}
          <span className="web-site-description-title-best">best</span> audio
          gear
        </h4>
        <p className="Body-manrope-medium web-site-description-text">
          Located at the heart of New York City, Audiophile is the premier store
          for high end headphones, earphones, speakers, and audio accessories.
          We have a large showroom and luxury demonstration rooms available for
          you to browse and experience a wide range of our products. Stop by our
          store to meet some of the fantastic people who make Audiophile the
          best place to buy your portable audio equipment.
        </p>
      </div>
    </div>
  );
};

export default WebSiteDescription;
