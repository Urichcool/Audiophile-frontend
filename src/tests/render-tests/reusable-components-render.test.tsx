import { renderWithReduxAndRouter } from "../helpers/RenderWithReduxAndRouter";
import CategoriesList from "../../components/Reusable-Components/CategoriesList";
import { screen } from "@testing-library/react";
import WebSiteDescription from "../../components/Reusable-Components/WebSiteDescription";
import CategoriesProductsListItem from "../../components/Reusable-Components/CategoriesProductsListItem";

describe("reusable-components-render-tests", () => {
  test("categories-list-should-renders-correctly", () => {
    renderWithReduxAndRouter(<CategoriesList isMobile={false} />);
    const categoriesList: HTMLUListElement =
      screen.getByTestId("categories-list");
    expect(categoriesList).toMatchSnapshot();
  });
  test("web-site-description-should-renders-correctly", () => {
    renderWithReduxAndRouter(<WebSiteDescription />);
    const webSiteDescription: HTMLDivElement = screen.getByTestId(
      "web-site-description"
    );
    expect(webSiteDescription).toMatchSnapshot();
  });
  test("categories-product-list-item-should-renders-correctly", () => {
    const {
      previewImage,
      isNew,
      name,
      description,
      isEven,
      id,
    }: {
      previewImage: { mobile: string; tablet: string; desktop: string };
      isNew: boolean;
      name: string;
      description: string;
      isEven: boolean;
      id: string;
    } = {
      previewImage: {
        mobile:
          "https://res.cloudinary.com/dir4brsjf/image/upload/v1699443154/audiophile/goods/product-xx99-mark-two-headphones/mobile/image-product-removebg-mobile-removebg-preview_e3ysuj.png",
        tablet:
          "https://res.cloudinary.com/dir4brsjf/image/upload/v1699443187/audiophile/goods/product-xx99-mark-two-headphones/tablet/image-product-removebg-tablet-removebg-preview_wbyhs5.png",
        desktop:
          "https://res.cloudinary.com/dir4brsjf/image/upload/v1699443139/audiophile/goods/product-xx99-mark-two-headphones/desktop/image-product-removebg-desktop-removebg-preview_myfyr1.png",
      },
      isNew: true,
      name: "XX99 Mark II Headphones",
      description:
        "The new XX99 Mark II headphones is the pinnacle of pristine audio. It redefines your premium headphone experience by reproducing the balanced depth and precision of studio-quality sound.",
      isEven: false,
      id: "652bdfecc2dd2dacebf6e267",
    };
    renderWithReduxAndRouter(
      <CategoriesProductsListItem
        previewImage={previewImage}
        isNew={isNew}
        name={name}
        description={description}
        isEven={isEven}
        id={id}
      />
    );
    const categoriesListItem: HTMLLIElement = screen.getByTestId(
      "categories-products-list-item"
    );
    expect(categoriesListItem).toMatchSnapshot();
  });
});
