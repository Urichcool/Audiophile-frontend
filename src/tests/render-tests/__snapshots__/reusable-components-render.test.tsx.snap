// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reusable-components-render-tests categories-list-component-should-renders-correctly 1`] = `
<ul
  class="categories-list"
  data-testid="categories-list"
>
  <li
    class="categories-list-item"
  >
    <div
      class="categories-list-hadphones-image"
    />
    <div
      class="categories-list-icon"
    >
      <svg
        fill="none"
        height="102"
        viewBox="0 0 183 102"
        width="183"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          filter="url(#filter0_f_0_4858)"
        >
          <ellipse
            cx="91.5001"
            cy="51"
            fill="black"
            rx="47.4444"
            ry="7"
          />
        </g>
        <defs>
          <filter
            color-interpolation-filters="sRGB"
            filterUnits="userSpaceOnUse"
            height="100.985"
            id="filter0_f_0_4858"
            width="181.874"
            x="0.563156"
            y="0.507492"
          >
            <feflood
              flood-opacity="0"
              result="BackgroundImageFix"
            />
            <feblend
              in="SourceGraphic"
              in2="BackgroundImageFix"
              mode="normal"
              result="shape"
            />
            <fegaussianblur
              result="effect1_foregroundBlur_0_4858"
              stdDeviation="21.7463"
            />
          </filter>
        </defs>
      </svg>
    </div>
    <h6
      class="H6-manrope-bold categories-list-item-title"
    >
      headphones
    </h6>
    <button
      class="button3"
      data-testid="navigation-button-headphones"
    >
      <p>
        Shop
      </p>
      <svg
        fill="none"
        height="12"
        viewBox="0 0 7 12"
        width="7"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1.3219 1L6.3219 6L1.3219 11"
          stroke="#D87D4A"
          stroke-width="2"
        />
      </svg>
    </button>
  </li>
  <li
    class="categories-list-item"
  >
    <div
      class="categories-list-speakers-image"
    />
    <div
      class="categories-list-icon"
    >
      <svg
        fill="none"
        height="102"
        viewBox="0 0 183 102"
        width="183"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          filter="url(#filter0_f_0_4858)"
        >
          <ellipse
            cx="91.5001"
            cy="51"
            fill="black"
            rx="47.4444"
            ry="7"
          />
        </g>
        <defs>
          <filter
            color-interpolation-filters="sRGB"
            filterUnits="userSpaceOnUse"
            height="100.985"
            id="filter0_f_0_4858"
            width="181.874"
            x="0.563156"
            y="0.507492"
          >
            <feflood
              flood-opacity="0"
              result="BackgroundImageFix"
            />
            <feblend
              in="SourceGraphic"
              in2="BackgroundImageFix"
              mode="normal"
              result="shape"
            />
            <fegaussianblur
              result="effect1_foregroundBlur_0_4858"
              stdDeviation="21.7463"
            />
          </filter>
        </defs>
      </svg>
    </div>
    <h6
      class="H6-manrope-bold categories-list-item-title"
    >
      speakers
    </h6>
    <button
      class="button3"
      data-testid="navigation-button-speakers"
    >
      <p>
        Shop
      </p>
      <svg
        fill="none"
        height="12"
        viewBox="0 0 7 12"
        width="7"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1.3219 1L6.3219 6L1.3219 11"
          stroke="#D87D4A"
          stroke-width="2"
        />
      </svg>
    </button>
  </li>
  <li
    class="categories-list-item"
  >
    <div
      class="categories-list-earphones-image"
    />
    <div
      class="categories-list-icon"
    >
      <svg
        fill="none"
        height="102"
        viewBox="0 0 183 102"
        width="183"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          filter="url(#filter0_f_0_4858)"
        >
          <ellipse
            cx="91.5001"
            cy="51"
            fill="black"
            rx="47.4444"
            ry="7"
          />
        </g>
        <defs>
          <filter
            color-interpolation-filters="sRGB"
            filterUnits="userSpaceOnUse"
            height="100.985"
            id="filter0_f_0_4858"
            width="181.874"
            x="0.563156"
            y="0.507492"
          >
            <feflood
              flood-opacity="0"
              result="BackgroundImageFix"
            />
            <feblend
              in="SourceGraphic"
              in2="BackgroundImageFix"
              mode="normal"
              result="shape"
            />
            <fegaussianblur
              result="effect1_foregroundBlur_0_4858"
              stdDeviation="21.7463"
            />
          </filter>
        </defs>
      </svg>
    </div>
    <h6
      class="H6-manrope-bold categories-list-item-title"
    >
      earphones
    </h6>
    <button
      class="button3"
      data-testid="navigation-button-earphones"
    >
      <p>
        Shop
      </p>
      <svg
        fill="none"
        height="12"
        viewBox="0 0 7 12"
        width="7"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1.3219 1L6.3219 6L1.3219 11"
          stroke="#D87D4A"
          stroke-width="2"
        />
      </svg>
    </button>
  </li>
</ul>
`;

exports[`reusable-components-render-tests categories-product-list-item-component-should-renders-correctly 1`] = `
<li
  class="categories-products-list-item"
  data-testid="categories-products-list-item"
>
  <div
    class="categories-products-list-item-picture-container"
  >
    <div
      class="animate__animated"
      style="animation-duration: 1s; opacity: 0;"
    >
      <picture
        class=""
        data-testid="picture-component"
      >
        <source
          height="300"
          media="(min-width: 1440px)"
          srcset="https://res.cloudinary.com/dir4brsjf/image/upload/v1699443139/audiophile/goods/product-xx99-mark-two-headphones/desktop/image-product-removebg-desktop-removebg-preview_myfyr1.png"
          width="270"
        />
        <source
          height="250"
          media="(min-width: 768px)"
          srcset="https://res.cloudinary.com/dir4brsjf/image/upload/v1699443187/audiophile/goods/product-xx99-mark-two-headphones/tablet/image-product-removebg-tablet-removebg-preview_wbyhs5.png"
          width="220"
        />
        <img
          alt="XX99 Mark II Headphones"
          height="180"
          src="https://res.cloudinary.com/dir4brsjf/image/upload/v1699443154/audiophile/goods/product-xx99-mark-two-headphones/mobile/image-product-removebg-mobile-removebg-preview_e3ysuj.png"
          width="160"
        />
      </picture>
    </div>
  </div>
  <div
    class="animate__animated"
    style="animation-duration: 1s; opacity: 0;"
  >
    <div
      class="categories-products-list-item-content-container"
    >
      <p
        class="Overline-manrope-regular categories-products-list-item-new-product-text"
      >
        NEW PRODUCT
      </p>
      <h2
        class="H2-manrope-bold categories-products-list-item-title"
      >
        XX99 Mark II Headphones
      </h2>
      <p
        class="Body-manrope-medium categories-products-list-item-description"
      >
        The new XX99 Mark II headphones is the pinnacle of pristine audio. It redefines your premium headphone experience by reproducing the balanced depth and precision of studio-quality sound.
      </p>
      <button
        class="button1"
        data-testid="id-page-nav-button"
      >
        See product
      </button>
    </div>
  </div>
</li>
`;

exports[`reusable-components-render-tests picture-component-should-renders-correctly 1`] = `
<picture
  class="hero-new-product-picture"
  data-testid="picture-component"
>
  <source
    height="300"
    media="(min-width: 1440px)"
    srcset="https://res.cloudinary.com/dir4brsjf/image/upload/v1699443139/audiophile/goods/product-xx99-mark-two-headphones/desktop/image-product-removebg-desktop-removebg-preview_myfyr1.png"
    width="270"
  />
  <source
    height="250"
    media="(min-width: 768px)"
    srcset="https://res.cloudinary.com/dir4brsjf/image/upload/v1699443187/audiophile/goods/product-xx99-mark-two-headphones/tablet/image-product-removebg-tablet-removebg-preview_wbyhs5.png"
    width="220"
  />
  <img
    alt="XX99 Mark II Headphones"
    height="180"
    src="https://res.cloudinary.com/dir4brsjf/image/upload/v1699443154/audiophile/goods/product-xx99-mark-two-headphones/mobile/image-product-removebg-mobile-removebg-preview_e3ysuj.png"
    width="160"
  />
</picture>
`;

exports[`reusable-components-render-tests web-site-description-component-should-renders-correctly 1`] = `
<div
  class="web-site-description-container"
  data-testid="web-site-description"
>
  <div
    class="web-site-description-picture"
  />
  <div
    class="web-site-description-text-container"
  >
    <h4
      class="H4-manrope-bold web-site-description-title"
    >
      Bringing you the
       
      <span
        class="web-site-description-title-best"
      >
        best
      </span>
       audio gear
    </h4>
    <p
      class="Body-manrope-medium web-site-description-text"
    >
      Located at the heart of New York City, Audiophile is the premier store for high end headphones, earphones, speakers, and audio accessories. We have a large showroom and luxury demonstration rooms available for you to browse and experience a wide range of our products. Stop by our store to meet some of the fantastic people who make Audiophile the best place to buy your portable audio equipment.
    </p>
  </div>
</div>
`;
